src/ext/sgx/mod.rs

```diff
@@ -1,47 +1,44 @@
+mod quote;
+
 use crate::crypto::*;
 
 use std::fmt::Debug;
 
-use anyhow::{anyhow, Result};
-
+use anyhow::{anyhow, Context, Result};
+use const_oid::db::rfc5912::ECDSA_WITH_SHA_256;
 use const_oid::ObjectIdentifier;
 use der::{Decodable, Sequence};
-use x509::ext::Extension;
-use x509::{request::CertReqInfo, Certificate};
-use x509::{PkiPath, TbsCertificate};
+use pkcs8::AlgorithmIdentifier;
+use quote::Quote;
+use x509::{ext::Extension, request::CertReqInfo, Certificate, TbsCertificate};
 
 use super::ExtVerifier;
 
 #[derive(Clone, Debug, PartialEq, Eq, Sequence)]
 pub struct Evidence<'a> {
-    pub pck: Certificate<'a>,
-
     #[asn1(type = "OCTET STRING")]
     pub quote: &'a [u8],
 }
```

Comment:
If we truly only have the quote, then we can throw away the `Evidence` structure and just get the quote directly from the extension encoded bytes.